src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx

```diff
@@ -0,0 +1,121 @@
+import { WorkspaceForm, WorkspaceFormSubmitData, WorkspaceOperationType } from '../workspace_form';
+import { WORKSPACE_OVERVIEW_APP_ID } from '../../../common/constants';
+import { formatUrlWithWorkspaceId } from '../../../../../core/public/utils';
+import { WorkspaceClient } from '../../workspace_client';
+import { WorkspaceFormData } from '../workspace_form/types';
+
+function getFormDataFromWorkspace(
+  currentWorkspace: WorkspaceAttribute | null | undefined
+): WorkspaceFormData {
+  return (currentWorkspace || {}) as WorkspaceFormData;
+}
+
+export const WorkspaceUpdater = () => {
+  const {
+    services: { application, workspaces, notifications, http, workspaceClient },
+  } = useOpenSearchDashboards<{ workspaceClient: WorkspaceClient }>();
+
+  const currentWorkspace = useObservable(workspaces ? workspaces.currentWorkspace$ : of(null));
+  const [currentWorkspaceFormData, setCurrentWorkspaceFormData] = useState<WorkspaceFormData>(
+    getFormDataFromWorkspace(currentWorkspace)
+  );
+
+  useEffect(() => {
+    setCurrentWorkspaceFormData(getFormDataFromWorkspace(currentWorkspace));
+  }, [currentWorkspace]);
+
+  const handleWorkspaceFormSubmit = useCallback(
+    async (data: WorkspaceFormSubmitData) => {
+      let result;
+      if (!currentWorkspace) {
+        notifications?.toasts.addDanger({
+          title: i18n.translate('Cannot find current workspace', {
+            defaultMessage: 'Cannot update workspace',
+          }),
+        });
+        return;
+      }
+
+      try {
+        const { ...attributes } = data;
+        result = await workspaceClient.update(currentWorkspace.id, attributes);
+      } catch (error) {
+        notifications?.toasts.addDanger({
+          title: i18n.translate('workspace.update.failed', {
+            defaultMessage: 'Failed to update workspace',
+          }),
+          text: error instanceof Error ? error.message : JSON.stringify(error),
+        });
+        return;
+      }
+      if (result?.success) {
+        notifications?.toasts.addSuccess({
+          title: i18n.translate('workspace.update.success', {
+            defaultMessage: 'Update workspace successfully',
+          }),
+        });
+        if (application && http) {
+          // Redirect page after one second, leave one second time to show update successful toast.
+          window.setTimeout(() => {
+            window.location.href = formatUrlWithWorkspaceId(
+              application.getUrlForApp(WORKSPACE_OVERVIEW_APP_ID, {
+                absolute: true,
+              }),
+              currentWorkspace.id,
+              http.basePath
+            );
+          }, 1000);
+        }
+        return;
+      }
+      notifications?.toasts.addDanger({
+        title: i18n.translate('workspace.update.failed', {
+          defaultMessage: 'Failed to update workspace',
+        }),
+        text: result?.error,
+      });
+    },
```

Comment:
Thanks for this change. Looks like the error handling is duplicate, what if we change it like this:

```suggestion
      try {
        const { ...attributes } = data;
        result = await workspaceClient.update(currentWorkspace.id, attributes);
        if (result?.success) {
          notifications?.toasts.addSuccess({
            title: i18n.translate('workspace.update.success', {
              defaultMessage: 'Update workspace successfully',
            }),
          });
          if (application && http) {
            // Redirect page after one second, leave one second time to show update successful toast.
            window.setTimeout(() => {
              window.location.href = formatUrlWithWorkspaceId(
                application.getUrlForApp(WORKSPACE_OVERVIEW_APP_ID, {
                  absolute: true,
                }),
                currentWorkspace.id,
                http.basePath
              );
            }, 1000);
          }
          return;
        } else {
          throw Error('workspace.update.success failed')
        }
      } catch (error) {
        notifications?.toasts.addDanger({
          title: i18n.translate('workspace.update.failed', {
            defaultMessage: 'Failed to update workspace',
          }),
          text: error instanceof Error ? error.message : JSON.stringify(error),
        });
        return;
      }
```

The disadvantage in the suggestion change is, we also include ```window.setTimeout()``` and other window API call into the error handling, which for me it's ok :)

What do you think?

Reply:
Thanks for your review, the suggested change looks good to me, and I've changed the code in both update workspace page and create workspace page, please help to take a look.